insights/parsers/partitions.py

```diff
@@ -0,0 +1,60 @@
+"""
+Partitions - file ``/proc/partitions``
+======================================
+
+This parser reads the ``/proc/partitions`` file, which contains
+partition block allocation information.
+
+"""
+from insights import Parser, parser
+from insights.parsers import SkipException, parse_delimited_table
+from insights.specs import Specs
+
+
+@parser(Specs.partitions)
+class Partitions(Parser):
+    """A parser class for parsing the ``/proc/partitions`` file.
+
+    Sample input::
+
+        major minor  #blocks  name
+
+           3     0   19531250 hda
+           3     1     104391 hda1
+           3     2   19422585 hda2
+         253     0   22708224 dm-0
+         253     1     524288 dm-1
+
+    Examples:
+       >>> type(partitions)
+       <class 'insights.parsers.partitions.Partitions'>
+       >>> partitions.data[0].get('name')
+       'hda'
+       >>> partitions.data[0].get('major')
+       '3'
+       >>> partitions.get_blocks('dm-0')
+       '22708224'
+
+    Raises:
+        SkipException: When input is empty.
+    """
+    def parse_content(self, content):
+        if not content:
+            raise SkipException('Empty content')
+
+        self.data = parse_delimited_table(content,
+                                          heading_ignore=['major', 'minor'],
+                                          header_substitute=[('#blocks', 'blocks')])
+
+    def get_blocks(self, name):
+        """Returns block allocation value for provided partition name.
+
+        Args:
+            name (str): Name of the partition.
+
+        Returns:
+            string: Returns `blocks` if `name` present. Otherwise None.
+        """
+        for partition in self.data:
+            if partition.get('name', '') == name:
+                return partition.get('blocks', '')
```

Comment:
This is a little confusing to me, you are implicitly returning `None` if the partition is not in the list, but you are returning an empty string if the name is present and the `blocks` field is not present?  Keep in mind how a developer will use this function, the way it is written it will require a check for both `''` and `is not None`. You might want to make the return values consistent.

You might also want to consider raising a `KeyError` here if the name is not present to avoid any ambiguity about what is happening and be consistent with the python std modules.

As mentioned in the previous comment, having a dict of names would simplify this logic as well.

Reply:
This function is not really necessary now. It is now easy to get `blocks` field for a partition.
```python
partitions_obj['sdb'].get('blocks')
```